csrc/ops/composite.cpp

```diff
@@ -89,6 +89,40 @@ TensorView* matmul(TensorView* a, TensorView* b) {
   return matmul(a, b, true /* cast output to input dtype */);
 }
 
+TensorView* linear(TensorView* a, TensorView* b, TensorView* bias) {
+  NVF_CHECK(
+      a->nDims() == b->nDims(),
+      "The number of dimension of A and B do not match");
+  NVF_CHECK(a->nDims() == 2, "Only 2-D A/B Tensors are supported in Linear!");
+
+  std::vector<bool> bcast_dims(a->nDims() + 1, false);
+  // A: [M, Bcast, K]
+  // B: [Bcast, N, K]
+  bcast_dims.at(bcast_dims.size() - 2) = true;
+  auto* tv0b = broadcast(a, bcast_dims);
+  bcast_dims.at(bcast_dims.size() - 2) = false;
+  bcast_dims.at(bcast_dims.size() - 3) = true;
+  auto* tv1b = broadcast(b, bcast_dims);
+
+  NVF_CHECK(
+      a->getDataType().value() == b->getDataType().value(),
+      "data types of inputs to matmul don't match");
+
+  auto* output = fusedMultiplySum(tv0b, tv1b, {-1});
+  if (bias) {
+    NVF_CHECK(bias->nDims() == 1, "Only 1-D Bias are supported in Linear!");
+    auto* bias_with_cast = maybeCastOp(output->getDataType().value(), bias);
+    auto* bcast_bias = broadcast(bias_with_cast, {true, false});
+    auto* bias_output = add(output, bcast_bias);
+    return maybeCastOp(a->getDataType().value(), bias_output);
+  }
+  return maybeCastOp(a->getDataType().value(), output);
+}
+
+TensorView* linear(TensorView* a, TensorView* b) {
```

Comment:
Why is this not handled by optional argument `bias = nullptr` in the function above?

Reply:
The main reasons for this are here:
https://github.com/NVIDIA/Fuser/pull/2090#discussion_r1569150816

But I had tried std::optional, but didn't consider a lambda as you mentioned below. I'll try out the lambda.